pkg/crc/preflight/preflight_checks_tray_windows.go

```diff
@@ -19,126 +16,82 @@ import (
 	}
-	binPathWithArgs := fmt.Sprintf("%s daemon", strings.TrimSpace(binPath))
 
-	// get the password from user
-	password, err := input.PromptUserForSecret("Enter account login password for service installation", "This is the login password of your current account, needed to install the daemon service")
+	// Create the cmd file to start the Posh script at login
+	daemonBatchFileContent := fmt.Sprintf(`%s -file "%s"`, powershell.LocatePowerShell(), constants.DaemonPSscriptPath)
+	err = ioutil.WriteFile(constants.DaemonBatchFilePath, []byte(daemonBatchFileContent), 0600)
 	if err != nil {
-		return fmt.Errorf("Unable to get login password: %v", err)
-	}
-
-	// sanitize password
-	password = escapeWindowsPassword(password)
-
-	psScriptContent := genTrayInstallScript(
-		password,
-		tempDir,
-		binPathWithArgs,
-		constants.TrayExecutablePath,
-		constants.TrayShortcutName,
-		constants.DaemonServiceName,
-	)
-	psFilePath := filepath.Join(tempDir, "trayInstallation.ps1")
-
-	// write temporary ps script
-	if err = writePsScriptContentToFile(psScriptContent, psFilePath); err != nil {
-		return err
+		return fmt.Errorf("Failed to create required batch file for the daemon: %w", err)
 	}
 
-	// invoke the ps script
-	_, _, err = powershell.ExecuteAsAdmin("Installing System Tray for CodeReady Containers", psFilePath)
-	// wait for the script to finish executing
-	time.Sleep(2 * time.Second)
-	if err != nil {
-		logging.Debug("Failed to execute tray installation script")
-		return err
+	// Start the daemon process from the Posh script
+	if _, _, err := powershell.Execute(constants.DaemonPSscriptPath); err != nil {
+		return fmt.Errorf("Failed to start the daemon process: %w", err)
 	}
 
-	// check for 'success' file
-	if _, err = goos.Stat(filepath.Join(tempDir, "success")); goos.IsNotExist(err) {
-		return fmt.Errorf("Installation script didn't execute successfully: %v", err)
-	}
 	return nil
 }
 
-func escapeWindowsPassword(password string) string {
-	// escape specials characters (|`|$|"|') with '`' if present in password
-	if strings.Contains(password, "`") {
-		password = strings.ReplaceAll(password, "`", "``")
-	}
-	if strings.Contains(password, "$") {
-		password = strings.ReplaceAll(password, "$", "`$")
+func fixTrayInstalled() error {
+	/* Start the tray and daemon process and copy the executables to start-up folder
+	 * "$USERPROFILE\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Startup"
+	 */
+	if err := fixDaemonInstalled(); err != nil {
+		return fmt.Errorf("Failed to install CodeReady Containers daemon: %w", err)
 	}
-	if strings.Contains(password, "\"") {
-		password = strings.ReplaceAll(password, "\"", "`\"")
+	cmd := fmt.Sprintf(`New-Item -ItemType SymbolicLink -Path "%s" -Name "%s" -Value "%s"`,
+		constants.StartUpFolder,
+		constants.TrayShortcutName,
+		constants.TrayExecutablePath,
+	)
+	if _, _, err := powershell.ExecuteAsAdmin("Copying tray and daemon executable to start-up folder", cmd); err != nil {
```

Comment:
Does this have to be done as an admin? `err = ioutil.WriteFile(constants.DaemonBatchFilePath, []byte(daemonBatchFileContent), 0600)` earlier is in the same directory I believe, and did not require admin?

Reply:
Strangely it needs `ExecuteAsAdmin` to copy the file, it could be that the powershell cmdlet that is being used needs an elevated shell, will try with golang `os.Symlink` and if that works we can remove the `ExecuteAsAdmin`

Reply:
Yup, odd :-/